### PR TITLE
NODE-31 Deleting Documents. Initialization, Operation, Finalization.

### PR DESCRIPTION
**Pull request by AndreiPanti**
https://github.com/marklogic/node-client-api/issues/665

### CODE REVIEW

File: lib/documents.js [hunk 1]
```diff
@@ -3776,6 +3776,361 @@ function queryToTransformAllDocumentsImpl(query, jobOptions){
...
+ */
+Documents.prototype.removeAllUris = function removeAllUrisDocuments(options) {
+  return removeAllUrisDocumentsImpl.call(
+      this, options
+  );
+};
+/*
+the batchSize defaults to the number recommended by Performance Engineering for remove operations (possibly 250).
+the transformStrategy and transform properties are not supported.
+*/
```
Comment (anu3990) [inline lib/documents.js:3796]: Please remove this comment.

File: test-basic/documents-data-movement-removeAllUris.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    //
+    //     restAdminDB.config.transforms.write(removeName, 'javascript', fs.createReadStream(removePath))
+    //         .result(()=>{
+    //             for(let i=0; i<100; i++){
+    //                 uris.push('/test/dataMovement/requests/removeAllUris/'+i+'.json');
+    //             }
+    //         })
+    //         .then(()=> done());
+    // });
+
```
Comment (anu3990) [inline test-basic/documents-data-movement-removeAllUris.js:43]: Please remove this block if it is not needed.

File: test-basic/documents-data-movement-removeAllUris.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+const dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnection);
+const Stream = require('stream');
+const ctsQb = marklogic.ctsQueryBuilder;
+const q = marklogic.queryBuilder;
+const query = q.where(ctsQb.cts.directoryQuery('/test/dataMovement/requests/removeAllUris/'));
+
+let removeStream = new Stream.PassThrough({objectMode: true});
+let uris = [];
+let removeName = 'WriteBatcherTest_remove';
+let removePath = './test-basic/data/removeAllUris_remove.js';
```
Comment (anu3990) [inline test-basic/documents-data-movement-removeAllUris.js:29]: Please remove these two variables if they are not needed.

**Review by anu3990 — CHANGES_REQUESTED** [review]

**Review by anu3990 — APPROVED** [review]

**Comment by anu3990** [comment]
@AndreiPanti please combine everything into one commit before merging.

**Review by georgeajit — APPROVED** [review]

**Pull request merged**